### PR TITLE
Refactor FormEntryActivity

### PR DESCRIPTION
**Pull request by phillipm**
Code structure changes:
 - Make FormEntryActivity extend CommCareActivity. This makes it easier to manage form load and save progress bars between screen orientation changes.
 - Hook up task based progress dialog using CommCareTask / CommCareActivity framework
 - Remove question progress badge code
 - Remove end of form view
 - General linting
 - Move image manipulation to helper class
 - Move navigation ui to helper class

Probably best reviewing commit by commit, but realize that I clean some of my changes up in the last few commits. First handful of commits (from 2 weeks earlier) are purely lint guided.

### CODE REVIEW

File: app/src/org/odk/collect/android/activities/FormEntryActivity.java [hunk 1]
```diff
@@ -516,6 +438,7 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                 saveAnswersForCurrentScreen(DO_NOT_EVALUATE_CONSTRAINTS);
                 break;
             case HIERARCHY_ACTIVITY:
+            case HIERARCHY_ACTIVITY_FIRST_START:
```
Comment (phillipm) [inline app/src/org/odk/collect/android/activities/FormEntryActivity.java]: Needed for refreshing the UI when you open the hierarchy view for a saved for and select a question to jump to.

File: app/src/org/odk/collect/android/activities/FormEntryActivity.java [hunk 2]
```diff
@@ -1358,217 +868,68 @@ private String getHeaderString() {
         } else {
             return StringUtils.getStringRobust(this, R.string.app_name) + " > " + mFormController.getFormTitle();
         }
-
     }
 
     /**
      * Creates a view given the View type and an event
```
Comment (amstone326) [inline app/src/org/odk/collect/android/activities/FormEntryActivity.java]: This comment seems outdated

**Comment by phillipm** [comment]
@amstone326 Thanks for the good feedback, PR should be updated. I ended up taking out the indeterminate progress bar stuff because the `CustomProgressDialog` automatically added in a progress spinner if no progress bar was present.

File: app/src/org/commcare/android/tasks/templates/CommCareTask.java [hunk 3]
```diff
@@ -89,6 +94,12 @@ protected void onPostExecute(C result) {
 
     protected abstract void deliverError(R receiver, Exception e);
 
+    /**
+     * Allows signalling to connected receiver that the task has been cancelled
+     */
+    protected void deliverCancellation(R receiver) {
```
Comment (ctsims) [inline app/src/org/commcare/android/tasks/templates/CommCareTask.java]: What makes this specifically different from connector.taskCancelled?

**Comment by ctsims** [comment]
@phillipm  One comment, good to pull after

File: app/src/org/commcare/android/tasks/templates/CommCareTask.java (hunk 3 above)
Comment (phillipm) [inline app/src/org/commcare/android/tasks/templates/CommCareTask.java]: @ctsims  Oh I missed that! I've updated it to override `taskCancelled`.

**Pull request merged**